Gems/NvCloth/Code/Source/Pipeline/SceneAPIExt/ClothRule.cpp

```diff
@@ -35,9 +35,9 @@ namespace NvCloth
 
         AZStd::vector<AZ::Color> ClothRule::ExtractClothData(
             const AZ::SceneAPI::Containers::SceneGraph& graph,
+            const AZ::SceneAPI::Containers::SceneGraph::NodeIndex meshNodeIndex,
             const size_t numVertices) const
         {
-            const auto meshNodeIndex = graph.Find(GetMeshNodeName());
```

Comment:
This was the problem. `GetMeshNodeName()` always refers to the non-optimized mesh.

Reply:
Due to the reasons explained in the comment, Instead of removing this, please change it to something like this:
````
auto meshNodeIndex = graph.Find(GetMeshNodeName() + Utilities::OptimizedMeshSuffix);
if (!meshNodeIndex.IsValid())
{
    meshNodeIndex = graph.Find(GetMeshNodeName());
}
````
Revert the rest of changes.

Reply:
Fixed. @moraaar can you review the new revision and resolve this conversation if it looks good to you?

Reply:
Thanks for the changes. It looks good.